plugins/org.komodo.core/src/org/komodo/repository/PropertyDescriptorImpl.java

```diff
@@ -209,4 +215,42 @@ public boolean isMultiple() {
         return this.multiple;
     }
 
+    /**
+     * {@inheritDoc}
+     *
+     * @see org.komodo.spi.repository.PropertyDescriptor#isValid(java.lang.Object)
+     */
+    @Override
+    public boolean isValid( final Object proposedValue ) {
+        if ( ( this.constraints != null ) && ( this.constraints.length != 0 ) ) {
+            if ( Type.STRING == this.type ) {
+                String value = null;
+
+                if ( proposedValue instanceof String ) {
+                    value = ( String )proposedValue;
+                } else if ( proposedValue instanceof Value ) {
+                    try {
+                        value = ( ( Value )proposedValue ).getString();
+                    } catch ( final Exception e ) {
+                        return false;
+                    }
+                } else {
+                    value = proposedValue.toString();
+                }
+
+                for ( final String pattern : this.constraints ) {
+                    if ( value.matches( pattern ) ) {
+                        return true;
+                    }
+                }
+
+                return false;
+            }
+
+            // TODO implement other type validation
```

Comment:
Keep it simple. If it introduces lots of complexity and work and we don't really need it then get rid of it.